### PR TITLE
schematracker(dm): ensure no MockStore in future

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #5334

#### Tests 

 - Unit test

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
release-7.5 should be reverted in https://github.com/pingcap/tiflow/pull/10970

File: dm/pkg/schema/tracker.go [hunk 1]
```diff
@@ -147,12 +146,6 @@ func (tr *Tracker) Init(
 	}
 	// TODO: need to use upstream timezone to correctly check literal is in [1970, 2038]
 	sctx := mock.NewContext()
```
Comment (Benjamin2037) [inline dm/pkg/schema/tracker.go:148]: Can we also add a comment to avoid other developers adding this back?

**Review by Benjamin2037 — APPROVED** [review]
reest LGTM

File: dm/pkg/schema/tracker.go [hunk 2]
```diff
@@ -147,12 +146,6 @@ func (tr *Tracker) Init(
...
-	store, err := mockstore.NewMockStore()
-	if err != nil {
-		return err
-	}
-	sctx.Store = store
-	err = sctx.NewTxn(ctx)
 	if err != nil {
```
Comment (3AceShowHand) [inline dm/pkg/schema/tracker.go:149]: this error check should also be removed?

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**